frontend/packages/console-dynamic-plugin-sdk/src/extensions/console-types.ts

```diff
@@ -141,7 +143,7 @@ export type PrometheusResponse = {
 };
 
 export type WatchK8sResource = {
-  kind: K8sResourceKindReference;
+  kind: K8sResourceKindReference | K8sGroupVersionKind; // can use reference (old format) or K8sGroupVersionKind
```

Comment:
This is a really good time to fix the misnamed `kind` prop. Instead of using `|`, I'd add a separate `groupVersionKind` prop that only takes `K8sGroupVersionKind`. Then we can deprecate the `kind` prop entirely.

In the API, we should strive for `kind` to *only* mean the string `kind` of the resource as it appears in the k8s API (`Deployment`, `Pod`, etc.). The kind and reference stuff is a hold-over from a lot of the code being written before k8s had API groups.

Same comment for the other two types below.

Reply:
Thats a good point Sam... good opportunity to do that is now.

Reply:
Another option is to make `group`, `version`, and `kind` top level properties. (If `group` and `version` are missing, we assume `kind` is a reference.) Opinions?

Reply:
Personal opinion would be to have a single prop. 

Making multiple opens up to a more complex interface that I think is confusing to the user imo. "What happens if I don't pass x, y, or z -- they're optional... what if I only have x... what if I only have y". I think one prop that takes the object feels nicer. And we can have utilities to craft that so the use case is "call me with this utility in tow" and not "figure it out yourself, with or without our utility".

Reply:
>This is a really good time to fix the misnamed kind prop. Instead of using |, I'd add a separate groupVersionKind prop that only takes K8sGroupVersionKind. Then we can deprecate the kind prop entirely.

As part of this PR have updated this to have `groupVersionKind` separately along with `kind`. And added `deprecated` for the kind prop.

am confused with the decision on `kind` naming?   shall we have it as `kind` for now as it takes a mix of both i.e k8sKind and k8sKindReference (group, version, king separated with ~)?